src/v/pandaproxy/schema_registry/sharded_store.cc

```diff
@@ -495,6 +495,43 @@ ss::future<bool> sharded_store::delete_subject_version(
       });
 }
 
+ss::future<mode> sharded_store::get_mode() {
+    co_return _store.local().get_mode().value();
+}
+
+ss::future<mode>
+sharded_store::get_mode(subject sub, default_to_global fallback) {
+    auto sub_shard{shard_for(sub)};
+    co_return co_await _store.invoke_on(
+      sub_shard, [sub{std::move(sub)}, fallback](store& s) {
+          return s.get_mode(sub, fallback).value();
+      });
+}
+
+ss::future<bool> sharded_store::set_mode(mode m, force f) {
+    auto map = [m, f](store& s) { return s.set_mode(m, f).value(); };
+    auto reduce = std::logical_and<>{};
+    co_return co_await _store.map_reduce0(map, true, reduce);
+}
+
+ss::future<bool>
+sharded_store::set_mode(seq_marker marker, subject sub, mode m, force f) {
+    auto sub_shard{shard_for(sub)};
+    co_return co_await _store.invoke_on(
+      sub_shard, _smp_opts, [marker, sub{std::move(sub)}, m, f](store& s) {
+          return s.set_mode(marker, sub, m, f).value();
```

Comment:
i guess our default `result<>` configuration has a policy that throws on `.value()` (rather than what I always assume to be undefined behavior if no value exists) ?

Reply:
It's [not default](https://github.com/BenPope/redpanda/blob/0ee456a83243d1aa17a7793c7b1b58f5d83b71d7/src/v/pandaproxy/schema_registry/errors.h#L23-L53); but it does throw.